tools/builder/ttir/ttir_utils.py

```
@@ -287,6 +347,14 @@ def run_ttir_pipeline(
         pipeline_options.append(f"mesh-shape={mesh_shape[0]},{mesh_shape[1]}")
     if argument_types_string:
         pipeline_options.append("enable-const-eval=true")
+        tt_populate_argument_types(module, argument_types_string)
+    if "TTMLIR_ENABLE_OPMODEL" in os.environ and (
```

we don't want to have optimizer APIs enabled by querying env flag. lets figure out how optimizer is built and if its not enabled, it should throw an error from their end.